classes/class-wpcom-liveblog-rest-api.php

```diff
@@ -0,0 +1,496 @@
+
+		self::set_liveblog_vars( $post_id );
+
+		// Get entry preview
+		$preview = WPCOM_Liveblog::format_preview_entry( $entry_content );
+
+		// Possibly do not cache the response
+		WPCOM_Liveblog::prevent_caching_if_needed();
+
+		return $preview;
+	}
+
+	/**
+	 * Get a list of authors matching a search term.
+	 *
+	 * @param WP_REST_Request $request A REST request object
+	 *
+	 * @return array An array of authors on the site
+	 */
+	public static function get_authors( WP_REST_Request $request ) {
+
+		// Get required parameters from the request
+		$term = $request->get_param( 'term' );
+
+		// Get a list of authors
+		$liveblog_authors = new WPCOM_Liveblog_Entry_Extend_Feature_Authors();
+		$authors = $liveblog_authors->get_authors( $term );
+
+		return $authors;
+	}
+
+	/**
+	 * Get a list of hashtags matching a search term
+	 *
+	 * @param WP_REST_Request $request A REST request object
+	 *
+	 * @return array An array of matching hastags
+	 */
+	public static function get_hash_terms( WP_REST_Request $request ) {
+
+		// Get required parameters from the request		
+		$term = $request->get_param( 'term' );
+
+		// Get a list of authors
+		$liveblog_hashtags = new WPCOM_Liveblog_Entry_Extend_Feature_Hashtags();
+		$hashtags = $liveblog_hashtags->get_hash_terms( $term );
+
+		return $hashtags;
+	}
+
+	/**
+	 * Set the Liveblog state of a post
+	 *
+	 * @param WP_REST_Request $request A REST request object
+	 *
+	 * @return string THe metabox markup to be displayed
+	 */
+	public static function update_post_state( WP_REST_Request $request ) {
+
+		// Get required parameters from the request		
+		$post_id         = $request->get_param( 'post_id' );
+		$state           = $request->get_param( 'state' );
+
+		// Additional request variables used in the liveblog_admin_settings_update action
+		$request_vars = array(
+			'state'                        => $state,
+			'liveblog-key-template-name'   => $request->get_param( 'template_name' ),
+			'liveblog-key-template-format' => $request->get_param( 'template_format' ),
+			'liveblog-key-limit'           => $request->get_param( 'limit' ),
+		);
+
+		self::set_liveblog_vars( $post_id );
+
+		// Get entry preview
+		$meta_box = WPCOM_Liveblog::admin_set_liveblog_state_for_post( $post_id, $state, $request_vars );
+
+		// Possibly do not cache the response
+		WPCOM_Liveblog::prevent_caching_if_needed();
+
+		return $meta_box;
+
+	}
+
+	/**
+	 * Set a few static variables in the WPCOM_Liveblog class needed for some callbacks to work
+	 *
+	 * @param int $post_id The post ID for the current request
+	 */
+	private static function set_liveblog_vars( $post_id ) {
+		WPCOM_Liveblog::$is_rest_api_call = true;
+		WPCOM_Liveblog::$post_id          = $post_id;
+	}
+
+	/**
+	 * Validation callback to check for numeric value
+	 *
+	 * @return bool true if $param is numeric. false otherwise.
+	 */
+	public static function validate_is_numeric( $param, $request, $key ) {
+		return is_numeric( $param );
```

Comment:
`is_numeric( -1 ) === true` so this may not the best check for our params since most (all?) expect positive integers.